tests/torchtune/data/test_messages.py

```diff
@@ -267,6 +296,55 @@ def test_call_train_on_input(self, sample):
         ]
         assert_dialogue_equal(actual["rejected"], expected_rejected)
 
+    @pytest.mark.parametrize(
+        "masking_strategy, expected_masks",
+        [
+            ("train_on_all", [[False, False], [False, False]]),
+            ("train_on_assistant", [[True, False], [True, False]]),
+            ("train_on_last", [[True, False], [True, False]]),
```

Comment:
Ideally we should explicitly test a case where the results of `train_on_assistant` and `train_on_last` differ

Reply:
I have added test cases using multi turn samples where results `train_on_assistant` and `train_on_last` differ where applicable (ChosenRejectedToMessages, ShareGPTToMessages, OpenAIToMessages)